File: Day6/lcp.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if(strs.length == 0 || strs == null){
            return "";
        }
        if (strs.length == 1)
            return strs[0];
        
        String prefix = strs[0];
        
        for (int i = 0; i < prefix.length(); i++){
            char check = prefix.charAt(i);
            
            for (int j = 1; j < strs.length; j++){
                if(i == strs[j].length() || strs[j].charAt(i) != check){
                    prefix = prefix.substring(0,i);
                }
            }
        }
        return prefix;
        
    }
}

File: Day6/lcp.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        
        if(strs.size() == 0)
            return "";
        
        if(strs.size() == 1 )
            return strs[0];
        
        string prefix = strs[0];
        
        for(int i = 0; i < prefix.size(); i++){
            
            char check = prefix.at(i);
            for (int j = 1; j < strs.size(); j++){
                if(i == strs[j].size() || strs[j].at(i) != check){
                    prefix = prefix.substr(0, i);
                }
            }
        }
        return prefix;
        
    }
};
